src/encord_active/frontend/src/components/explorer/Explorer.tsx

```diff
@@ -131,6 +146,48 @@ export function Explorer({
     setShowAnnotations(selectedMetric.domain === "annotation");
   }, [selectedMetric.domain, setShowAnnotations]);
 
+  // Data or Label selection
+  const analysisDomainOptions = useMemo(
+    () =>
+      Object.entries(AnalysisDomain).map(([key, value]) => ({
+        label: Object.prototype.hasOwnProperty.call(
+          analysisDomainLabelOverrides,
+          value
+        )
+          ? analysisDomainLabelOverrides[value as AnalysisDomain]
+          : key,
+        value,
+      })),
+    []
+  );
+  const [analysisDomain, setAnalysisDomain] = useState<AnalysisDomain>("data");
+  const [selectedMetricData, setSelectedMetricData] = useState<Metric>({
+    domain: "data",
+    metric_key: "metric_random",
+  });
+  const [selectedMetricLabel, setSelectedMetricLabel] = useState<Metric>({
+    domain: "annotation",
+    metric_key: "metric_random",
+  });
+
+  const handleSelectedDomainChange = (val: AnalysisDomain) => {
```

Comment:
selectedMetric could be a derived value (doesn't even need memo).

const selctedMetric = analysisDomain == "data ? selectedMetricData. : selectedMetricLabel